hs_core/tasks.py

```diff
@@ -363,6 +378,17 @@ def manage_task_hourly():
         # send email for people monitoring and follow-up as needed
         send_mail(subject, email_msg, settings.DEFAULT_FROM_EMAIL, [settings.DEFAULT_SUPPORT_EMAIL])
 
+    # update crossref deposit for published resource for which relevant metadata has been updated
+    update_deposit_resources = (BaseResource.objects
+                                .filter(raccess__published=True,
+                                        extra_data__contains={CrossRefSubmissionStatus.UPDATE: 'True'})
+                                .exclude(doi__contains=CrossRefSubmissionStatus.PENDING)
+                                )
+    for res in update_deposit_resources:
+        _update_crossref_deposit(res)
+        res.extra_data[CrossRefSubmissionStatus.UPDATE] = 'False'
```

Comment:
Similarly, let's look at a different scenario:
1. Author updates funding source on a published resource that has doi NO LONGER PENDING.
2. `extra_data.CROSSREF_UPDATE` gets set to True here https://github.com/hydroshare/hydroshare/blob/5225-improved-crossref-metadata-deposit/hs_core/models.py#L4390
3. In that case, the doi has no extra flags
4. Then the hourly task picks up the resource: https://github.com/hydroshare/hydroshare/blob/5225-improved-crossref-metadata-deposit/hs_core/tasks.py#L382-L390 and the `res.extra_data[CrossRefSubmissionStatus.UPDATE] = 'False'` is set.
5. Let's say that the deposit fails. In that case, the doi get a `failed` flag here https://github.com/hydroshare/hydroshare/blob/5225-improved-crossref-metadata-deposit/hs_core/tasks.py#L1217
6. Then the next hour, that results in the incorrect message going to the admin. Because CROSSREF_UPDATE = False and doi has `failed`, they get the ["failed deposit" message ](https://github.com/hydroshare/hydroshare/blob/5225-improved-crossref-metadata-deposit/hs_core/tasks.py#L308-L311) instead of the "failed update deposit" message, even though this was a metadata update.

Reply:
> Just want to check my understanding. Is the following true?
res.extra_data[CrossRefSubmissionStatus.UPDATE] can only be true if a res.doi is pending or failed

No. Resource doi doesn't need to be in pending or failed state. This is set any time we are updating crossref deposit for a published resource.

Reply:
> ...I guess this will ONLY happen if the original deposit is in a failed/pending state and the user tries to update the metadata for that published resource. ...

I don't think that's true. I think the wrong message will be sent even if the original deposit is no longer pending/failed. I outlined that scenario here:

https://github.com/hydroshare/hydroshare/pull/5274#discussion_r1417506515

I think the wrong message will be sent **any time that the update deposit fails**

I'm not too excited about adding another flag to track original vs not... :(

Reply:
https://github.com/hydroshare/hydroshare/pull/5274#discussion_r1417506515

The above scenario will work the following way:
1. Author updates funding source on a published resource that has doi NO LONGER PENDING.
2. extra_data.CROSSREF_UPDATE gets set to **False** here https://github.com/hydroshare/hydroshare/blob/5225-improved-crossref-metadata-deposit/hs_core/models.py#L4386
3. In that case, the doi has no extra flags
4. The async update task will trigger here  https://github.com/hydroshare/hydroshare/blob/5225-improved-crossref-metadata-deposit/hs_core/models.py#L4387
5. Let's say that the deposit fails. In that case, the doi get a failed flag here https://github.com/hydroshare/hydroshare/blob/5225-improved-crossref-metadata-deposit/hs_core/tasks.py#L1217
6. Then the next hour, when the failed deposit is re-deposited and that also fails then this condition https://github.com/hydroshare/hydroshare/blob/5225-improved-crossref-metadata-deposit/hs_core/tasks.py#L308 will be FALSE. That will result in sending the [update deposit failed](https://github.com/hydroshare/hydroshare/blob/5225-improved-crossref-metadata-deposit/hs_core/tasks.py#L314) message.

Reply:
What if we just alter the messages to make them more general, instead of adding another flag? Just an idea. Probably the responsible thing to do is add another flag...

Reply:
The main purpose of the flag `CROSSREF_UPDATE` is NOT to generate different email messages. The flag is used to mark the resource so that the system can update crossref deposit at a later time (hourly celery task) primarily when the update can't be done immediately after the user updates the metadata of a published resource because the resource deposit in a pending state.

Reply:
yep thanks for the reminder.
I was just thinking into the future and putting myself into the shoes of the Poor Soul (amin) who receives the email saying:

> Metadata deposition with CrossRef for the published resource DOI **wooohoo.doi.yes** failed again after retry with first metadata deposition requested since **yesterday**

Poor Soul (admin) wants to figure out what the state of things is. They see a "pending" doi when they check in crossref. Then they go to the django db and see that indeed the `.doi` is appended with "pending"

I'm just wondering if it would be impossible to figure out why things were failing and what should be done to remedy. But I don't have a better suggestion either, because adding more flags just makes it more complicated and I don't know that future Poor Soul will have the awareness of said flags in order to ascertain what they indicate.

Reply:
I am not sure if the following would make the admin life bit easier when looking at the doi of a published resource:
1. For initial deposit pending we append the **doi** with `pending`
2. For initial deposit failure we append the **doi** with `failure`
3. For subsequent update deposit pending we append the **doi** with `update-pending`
4. For subsequent update deposit failure we append the **doi** with `update-failure`

Reply:
I like that idea @pkdash good thinking!

Reply:
This would allow us to change the UI if we wanted (right now we indicate that it is _pending_ in the UI, but if we want we could further distinguish between "update" pending and "initial deposit" pending
I think that would be helpful for the admin debugging issues as well